1. Escreva um programa que efetue a leitura de 20 valores numéricos e apresente no final o total do somatório e a média dos valores lidos

In [3]:
soma = 0
n = 20
# aqui eu vou usar soma como variável auxiliar pra ficar melhor

for i in range(n):
    num = float(input("Digite sua nota: "))
    soma += + num
    # para i, o ciclo vai se repetir n(20) vezes.
    # num vai ser a entrada dos dados, soma vai ser a soma, normal
    # depois de repetir o ciclo 20 vezes, ele vai sair do loop e vai mostrar a media e a soma

media = soma/n
print(soma)
print(media)

20.0
1.0


2. Escreva um programa que calcula a média de 30 alunos e informa a situação (reprovado, aprovado ou recuperação).

In [11]:
soma = 0
n = 30

for i in range (n):
    entrada = float(input('Digite a nota: '))
    soma += + entrada
    
media = soma/n

if media >= 7:
    print('Aluno aprovado com média {:.2}'.format(media))

    
else:
    if media >= 4:
        print('Aluno em recuperação com média {:.2}'.format(media))
    else:
        print('Aluno reprovado com média {:.2}'.format(media))

ValueError: could not convert string to float: ''

3. Escreva um programa para mostrar os números ímpares de 10 a 100.

In [12]:
for i in range (1,100,2):
    print(i)


1
3
5
7
9
11
13
15
17
19
21
23
25
27
29
31
33
35
37
39
41
43
45
47
49
51
53
55
57
59
61
63
65
67
69
71
73
75
77
79
81
83
85
87
89
91
93
95
97
99


4. Fazer um programa para ler o sexo de dez pessoas e mostrar a quantidade de pessoas do sexo masculino e do sexo feminino, separadamente.

In [16]:
feminino = 'F'
masculino = 'M'
somaMasculino = somaFeminino = 0
n = 3

for i in range (n):
    entrada = str(input('Digite F ou M: '))
    if entrada == feminino or entrada == 'f':
        somaFeminino += + 1
    elif entrada == masculino or entrada == 'm':
        somaMasculino += + 1

    else:
        print('Sexo inválido.')

print('{} mulheres e {} homens.'.format(somaFeminino, somaMasculino))

2 mulheres e 1 homens.


5. Escreva um programa que leia 10 números e informe o maior e o menor número.

In [6]:
lista = []
n = 3
for i in range (n):
    num = int(input('Digite os números: '))
    lista.append(num)

print ('Maior número da lista: ', max(lista)) ; print('Menor número da lista:', min(lista))

Maior número da lista:  3
Menor número da lista 1


6. Escreva um programa que mostre os números divisíveis por 5 de 1 a 100.

In [10]:
for i in range (5,105,5):
    print(i)
    

5
10
15
20
25
30
35
40
45
50
55
60
65
70
75
80
85
90
95
100


7. Escreva um programa que calcula o fatorial de um dado número N.

In [15]:
from math import factorial
entrada = int(input('Digite um número: '))
print('Fatorial de {} é {}'.format(entrada, factorial(entrada)))

            


Fatorial de 5 é 120



8. Faça um programa para somar os números pares positivos < 1000 e ao final escrever o resultado.

In [28]:
soma = 0

for i in range(2,15,2):

soma += i
print(soma)




IndentationError: expected an indented block after 'for' statement on line 3 (1098044149.py, line 4)

9. Faça um programa para calcular a área de N quadriláteros. Fórmula: Área = Lado * Lado.

In [33]:
n = int(input('Quantos quadriláteros você deseja calcular a área? '))

for i in range(n):
    lado = float(input('Qual o valor do lado? '))
    lado2 = float(input('Qual o valor do lado? '))
    area = lado * lado2
    print('A área do quadrilatero é: {:.2f}'.format(area))

KeyboardInterrupt: Interrupted by user

In [ ]:
# Kézia colocou na questão que é a área dos quadriláteros, porém a fórmula que ela colocou engloba apenas quadrados e retângulos, a fórmula é diferente pra
# calcular trapézio, losangulo, etc.

# vou fazer uma versão mais completa dessa questão depois.

10. Faça um programa para calcular a área de N Círculos. Fórmula: Área =  raio2  = 3,141592.

In [37]:
import math 
n = int(input('Quantos circulos você deseja calcular a área? '))

for i in range(n):
    raio = float(input('Qual o valor do raio? '))
    area = math.pi + raio * 2
    print('A área do círculo é: {:.2f}'.format(area))

A área do círculo é: 23.14
A área do círculo é: 13.14


11. Faça um programa para calcular um valor A elevado a um expoente B. Os valores A e B deverão ser lidos.
Não usar A** B e sim uma estrutura de repetição.

In [38]:
n = int(input('Quantos exponenciais você deseja calcular? '))

for i in range(n):
    A = int(input('Qual o valor da base? '))
    B = int(input('Qual o valor do expoente?'))
    if B == 0:
        print(1)
    else: 
        exp = A ** B
        print(exp)

25
1


12. Faça um programa para ler um valor X e calcular Y = X +2X+3X+4X+…+20X

In [44]:
x = int(input('Qual o valor de x? '))
y = 0

for i in range(1,21):
    y = x + (i) * x
    print(y)

4
6
8
10
12
14
16
18
20
22
24
26
28
30
32
34
36
38
40
42


13. Sendo H = 1 + 1/2 + 1/3 + 1/4 + ... + 1/N. Faça um programa para gerar e mostrar o número H. O número N será fornecido como entrada.

In [65]:
n = int(input('Qual o valor de N? '))
h = 1

for i in range(1,11):
    h = h + 1/(n)
    print('{:.2f}'.format(h))


1.50
2.00
2.50
3.00
3.50
4.00
4.50
5.00
5.50
6.00


14. Faça um programa para: 
a) Ler um valor x qualquer
b) Calcular Y = (x+1)+(x+2)+(x+3)+(x+4)+(x+5)+…(x+100).

In [79]:
x = int(input('Insira o valor da variaável X: '))
n = 0

for i in range(0,101):
    y += x + i

print('Y =', y)


Y = 5252


15. Fazer um programa que calcule e escreva a soma dos 50 primeiros termos da seguinte série:

In [98]:
dividendo = 1000
# esse é o valor de cima
soma = 0
for i in range(1,51):
    # aqui eu vou fazer a parte de baixo, ele vai de 1 a 50.
    soma += dividendo/i
    # a minha soma vai ser soma + dividendo/i
                    # (inicialmente 0) + 1000/i (que vai mudando de acordo com a progessão).
    dividendo -= 3 
    # ao final de cada operação ele vai tirar 3 unidades, dessa forma dá tudo certo (dividendo = dividendo - 3).

print(soma)

4362.702954344412


16. Faça um programa que recebe o número real x como entrada e devolva uma aproximação do arco tangente de x (em radianos) através da série:

1.57


17. Faça um programa que mostre os n termos da Série a seguir: 

S = 1/1 + 2/3 + 3/5 + 4/7 + 5/9 + ... + n/m.
 
Imprima no final a soma da série. 

In [ ]:
n = int(input('Digite N: '))
m = int(input('Digite M: '))




18. Cada espectador de um cinema respondeu a um questionário no qual constava sua idade e a sua opinião em relação ao filme: ótimo - 3, bom - 2, regular - 1. 
Faça um programa que receba a idade e a opinião de 15 espectadores, calcule e imprima:
a)A média das idades das pessoas que responderam ótimo;
b)A quantidade de pessoas que responderam regular;
c)A porcentagem de pessoas que responderam bom entre todos os espectadores analisados.

In [112]:
idade = 0
avaliação_regular = 0
avaliação_bom = 0
avaliação_otimo = 0

for i in range(1,16):
    espectador_1 = int(input('Qual a sua idade? '))
    espectador_2 = int(input('Avalie o filme (1 - regular | 2 - bom | 3 - ótimo): '))
    idade += espectador_1
    if espectador_2 == 1:
        avaliação_regular += 1
    elif espectador_2 == 2:
        avaliação_bom += 1
    else:
        avaliação_otimo += 1

media_idade = idade/15
print('Média de idade é {}. Nº de pessoas que avaliaram o serviço como regular: {}.'.format(media_idade, avaliação_regular))
print(f'Porcentagem que achou bom: {(avaliação_bom/15)*100:.0f}%')




Média de idade é 25.0. Nº de pessoas que avaliaram o serviço como regular: 1.
Porcentagem que achou bom: 33%


19. Faça um programa que receba dois números inteiros e gere os números inteiros que estão no intervalo compreendido por eles.

In [113]:
n1 = int(input('Digite o início do intervalo: '))
n2 = int(input('Digite o término do intervalo: '))

for i in range(n1,n2+1):
    print(i)

1
2
3
4
5


20. Desenvolva um gerador de tabuada, capaz de gerar a tabuada de qualquer número inteiro entre 1 a 10. O usuário deve informar de qual numero ele deseja ver a tabuada. A saída deve ser conforme o exemplo abaixo: 


Tabuada de 5:
5 X 1 = 5
5 X 2 = 10
...
5 X 10 = 50

In [115]:
n = int(input('Digite um número para gerar a tabuada: '))

for i in range(1,11):
    tabuada = n * i
    print('{} x {} = {}'.format(n,i,tabuada))

8 x 1 = 8
8 x 2 = 16
8 x 3 = 24
8 x 4 = 32
8 x 5 = 40
8 x 6 = 48
8 x 7 = 56
8 x 8 = 64
8 x 9 = 72
8 x 10 = 80


21. Faça um programa que peça um número inteiro e determine se ele é ou não um número primo. Um número primo é aquele que é divisível somente por ele mesmo e por 1.

In [120]:
numero = int(input('Digite um nº: '))
contador = 0

for i in range(1,numero+1):
    if numero % i == 0:
        contador += 1
    
if contador == 2 and num != 0:
    print('Este número é primo')

else:
    print('Não é primo.')

Este número é primo


In [116]:
n = int(input('Quantas vezes vc quer que diga eu te amo? '))

for i in range(n):
    print("te amo, zackply <3")

te amo, zackply <3
te amo, zackply <3
te amo, zackply <3
te amo, zackply <3
te amo, zackply <3


22. Faça um programa que peça para n pessoas a sua idade, ao final o programa devera verificar se a média de idade da turma varia entre 0 e 25,26 e 60 e maior que 60; e então, dizer se a turma é jovem, adulta ou idosa, conforme a média calculada.

In [8]:
pessoas = int(input('Quantas pessoas são? '))

for i in range(pessoas):
    idade = int(input('Qual a sua idade? '))
    idade += + idade
    
media = idade / pessoas

if idade >= 0 and idade <=25:
    print('média de idade é {:.2f}, logo é uma turma jovem'.format(media))

elif idade >= 26 and idade <= 60:
    print('média de idade é {:.2f}, logo é uma turma adulta'.format(media))

else:
    print('média de idade é {:.2f}, logo é uma turma idosa'.format(media))




média de idade é 30.00, logo é uma turma adulta


23. O Departamento Estadual de Meteorologia lhe contratou para desenvolver um programa que leia um conjunto indeterminado de temperaturas, e informe ao final a menor e a maior temperatura informada, bem como a média das temperaturas. 

In [4]:
x = True
cont = []
contador = 0

while x:
    temperatura = float(input('Digite uma temperatura: '))
    contador += + 1
    cont.append(temperatura)

    fator = int(input('Deseja adicionar mais alguma coisa? 1 - sim, 0 - não '))
    if fator == 0:
        x =  False
    else:
        x = True

soma = sum(cont)
maxima = max(cont)
minima = min(cont)
print('A média de temperatura é {}, com máxima de {} e mínima de {}'.format(soma,maxima,minima))


A média de temperatura é 88.0, com máxima de 50.0 e mínima de 38.0


24. Faça um programa que receba o valor de uma dívida e mostre uma tabela com os seguintes dados: valor da dívida, valor dos juros, quantidade de parcelas e valor da parcela. 
Os juros e a quantidade de parcelas seguem a tabela abaixo: 
Quantidade de Parcelas  % de Juros sobre o valor inicial da dívida
         1      			 0
	3   		    	10
	6       			15
	
Exemplo de saída do programa: 
Valor da Dívida  Valor dos Juros  Quantidade de Parcelas  Valor da Parcela
R$ 1.000,00              0                         1                                  R$  1.000,00
R$ 1.100,00           100                        3                                 R$    366,00
R$ 1.150,00           150                        6                                 R$    191,67

In [12]:
valorDivida = float(input('Digite o valor da sua dívida: '))
juros3 = valorDivida * 0.10
juros6 = valorDivida * 0.15
valorParcelas3 = juros3 + valorDivida
valorParcelas6 = juros6 + valorDivida
parcelas3 = valorParcelas3 / 3
parcelas6 = valorParcelas6 / 6

# Lembrar de procurar uma forma mais bonitinha de fazer a apresentação desse print.

print('VALOR DA DÍVIDA\n R${}\n R${}\n R${}\n'.format(valorDivida,valorParcelas3,valorParcelas6))
print('VALOR DO JUROS\n R$0\n R${:.0f}\n R${:.0f}\n'.format(juros3,juros6))
print('QUANTIDADE DE PARCELAS\n 1 vez\n 3 vezes\n 6 vezes\n')
print('VALOR DA PARCELA\n R${:.0f}\n R${:.2f}\n R${:.2f}\n'.format(valorDivida,parcelas3,parcelas6))

VALOR DA DÍVIDA
 R$1000.0
 R$1100.0
 R$1150.0

VALOR DO JUROS
 R$0
 R$100
 R$150

QUANTIDADE DE PARCELAS
 1 vez
 3 vezes
 6 vezes

VALOR DA PARCELA
 R$1000
 R$366.67
 R$191.67



25. Faça um programa que peça uma nota, entre zero e dez. Mostre uma mensagem caso o valor seja inválido e continue pedindo até que o usuário informe um valor válido. 

In [9]:
x = True

while x:
    nota = int(input('Informe uma nota de 1 a 10: '))

    if nota >= 1 and nota <= 10:
        print('A nota informada foi: {}'.format(nota))
        x = False
    else:
        print('Nota inválida!')
        x = True

Nota inválida!
A nota informada foi: 10



26. Uma grande firma deseja saber qual é o empregado mais recente e qual é o mais antigo. Desenvolver um programa para ler um número indeterminado de informações contendo o número do empregado e o número de meses de trabalho deste empregado e imprimir o mais recente e o mais antigo. Obs.: A última informação contém os dois números iguais a zero. Não existem dois empregados admitidos no mesmo mês.

In [ ]:
x = True
empregados = []
meses = []

while x:
    identificacao = int(input('Digite o nº de identificação do empregado: '))
    empregados.append(identificacao)

    tempo = int(input('Quantos meses ele trabalha? '))
    meses.append(tempo)

    confirmação = int(input('Deseja adicionar mais algum dado? 1 - para SIM | 0 - para NÃO: '))

    if confirmação == 0:
        x = False

    else:
        x = True

maxima = max(empregados)
minima = min()
print('O empregado mais antigo é o nº{} com {} meses')
print('O empregado mais novo é o nº{} com {} meses')

#travei na parte de como pegar o nº do empregado com q quantidade de meses correspondente
# mas achei na internet essa resposta, REVISAR DEPOIS

# L = []
# while True:

   # n, m = input('Numero e numero de meses: ').split()

   # n, m = int(n), int(m)

   # if n == 0 and m == 0: break

   # L.append([n, m])

# L = sorted(L, key = lambda meses : meses[1])

# print(L[:3])


27. A prefeitura de uma cidade fez uma pesquisa entre seus habitantes, coletando dados sobre o salário e número de filhos. A prefeitura deseja saber:
a)Média do salário da população;
b)Média do número de filhos;
c)Maior salário;
d)Percentual de pessoas com salário até R$250,00.
Desenvolver um programa para calcular e escrever o que foi pedido nos itens a, b, c e d. O final da leitura de dados se dará com a entrada de um salário negativo.

In [25]:
salarios = []
salarios250 = []
filhos = []
listaTOTAL = []
x = True

while x:
    salario = float(input('Digite o valor do salário: '))
    Qfilhos = int(input('Digite a quantidade de filhos: '))

    if salario < 0:
        print('Você inseriu um salário negativo. Too bad, boy.')
        break
    else:
        x = True
        filhos.append(Qfilhos)
        if salario <= 250:
            salarios250.append(salario)
        else:
            salarios.append(salario)
        
        cont = int(input('Deseja continuar inserindo dados? 1 - SIM | 0 - NÃO: '))
        if cont == 1:
            x = True
        else: 
            x = False

soma = sum(salarios) + sum(salarios250)
mediaSalarios = soma / (len(salarios) + len(salarios250))
mediaFilhos = sum(filhos) / len(filhos)
maxSalario = max(salarios)

print('MÉDIA SALÁRIO DA POPULAÇÃO: R${}'.format(mediaSalarios))
print('MÉDIA DE FILHOS: {:.0f}'.format(mediaFilhos))
print('MAIOR SALÁRIO DA PESQUISA: R${}'.format(maxSalario))



Você inseriu um salário negativo. Too bad, boy.


ZeroDivisionError: division by zero

28. Faça um programa que leia uma quantidade não determinada de números positivos. Calcule a quantidade de números pares e ímpares, a média de valores pares e a média geral dos números lidos. O número que encerrará a leitura será zero.

29. Escrever um programa que leia uma quantidade desconhecida de números e conte quantos deles estão nos seguintes intervalos: [0.25], [26,50], [51,75] e [76,100]. A entrada de dados deve terminar quando for lido um número negativo.

30. Supondo que a população de um país A seja da ordem de 80000 habitantes com uma taxa anual de crescimento de 3% e que a população de B seja 200000 habitantes com uma taxa de crescimento de 1.5%. Faça um programa que calcule e escreva o número de anos necessários para que a população do país A ultrapasse ou iguale a população do país B, mantidas as taxas de crescimento.


31. O cardápio de uma lanchonete é o seguinte: 

Especificação 	        Código 	 	Preço
Cachorro Quente 	     100     	R$ 1,20
Bauru Simples            101     	R$ 1,30
Bauru com ovo            102     	R$ 1,50
Hambúrguer               103     	R$ 1,20
Cheeseburguer            104     	R$ 1,30
Refrigerante             105     	R$ 1,00

Faça um programa que leia o código dos itens pedidos e as quantidades desejadas. Calcule e mostre o valor a ser pago por item (preço * quantidade) e o total geral do pedido. Considere que o cliente deve informar quando o pedido deve ser encerrado. 